kliff/transforms/configuration_transforms/graphs/generate_graph.py

```diff
@@ -0,0 +1,199 @@
+        self.energy = None
+        self.forces = None
+        self.n_layers = None
+        self.coords = None
+        self.images = None
+        self.species = None
+        self.z = None
+        self.contributions = None
+
+    def __inc__(self, key, value, *args, **kwargs):
+        if "index" in key or "face" in key:
+            return self.num_nodes
+        elif "contributions" in key:
+            return 2
+        elif "images" in key:
+            return torch.max(value) + 1
+        else:
+            return 0
+
+    def __cat_dim__(self, key, value, *args, **kwargs):
+        if "index" in key or "face" in key:
+            return 1
+        else:
+            return 0
+
+
+class KLIFFTorchGraphGenerator(ConfigurationTransform):
+    """
+    Generate a graph representation of a configuration. This generator will also save the
+    required parameters for porting the model over to KIM-API using TorchMLModelDriver.
+    The configuration file saved here will generate identical graphs at KIM-API runtime.
+    For porting the graph representation you also need to provide the TorchScript model file name.
+
+    Args:
+        species (list): List of species.
+        cutoff (float): Cutoff distance.
+        n_layers (int): Number of convolution layers.
+        as_torch_geometric_data (bool): If True, the graph will be returned as a Pytorch
+            Geometric Data object.
+        implicit_fingerprint_copying (bool): If True, the fingerprint will be copied to
+            the Configuration object's fingerprint attribute.
+    """
+
+    def __init__(
+        self,
+        species,
+        cutoff,
+        n_layers,
+        as_torch_geometric_data=False,
+        implicit_fingerprint_copying=False,
+    ):
+        super().__init__(implicit_fingerprint_copying=implicit_fingerprint_copying)
+        self.species = species
+        self.cutoff = cutoff
+        self.n_layers = n_layers
+        self.infl_dist = n_layers * cutoff
+        self._tg = graph_module
+        if as_torch_geometric_data and not torch_geometric_available():
+            raise ImportError("Torch geometric is not available")
+        self.as_torch_geometric_data = as_torch_geometric_data
+
+    def forward(self, configuration: "Configuration"):
+        """
+        Generate a graph representation of a configuration.
+
+        Args:
+            configuration: Instance of ~:class:`kliff.dataset.Configuration`. For which the
+                graph representation is to be generated.
+
+        Returns:
+            C++ custom graph object or Pytorch Geometric Data object.
+        """
+        graph = graph_module.get_complete_graph(
+            self.n_layers,
+            self.cutoff,
+            configuration.species,
+            configuration.coords,
+            configuration.cell,
+            configuration.PBC,
+        )
+        graph.energy = configuration.energy
+        graph.forces = configuration.forces
+        if torch_available:
```

Comment:
This `if torch_available` is error prone, for example, in the case a user wants to export to PyG graph, but forget to install torch.

Essentially, this should only be detemined by `self.as_torch_geometric_data`. If it is true, we should export a PyG graph. We can raise error if torch is not available.

Reply:
See above. Now `PyGGraph` is always used. If PyG is present then it will be an instance of `pyg.data.Data` otherwise it will be a plain Python class with same fields. In the latter case it will log warning as well.